packages/google_sign_in/google_sign_in/android/src/test/java/io/flutter/plugins/googlesignin/GoogleSignInTest.java

```diff
@@ -5,13 +5,188 @@
 package io.flutter.plugins.googlesignin;
 
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
+import android.app.Activity;
 import android.content.Context;
+import android.content.Intent;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.common.api.Scope;
 import io.flutter.plugin.common.BinaryMessenger;
 import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.PluginRegistry;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import org.junit.Before;
 import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
 
 public class GoogleSignInTest {
+  @Mock Context mockContext;
+  @Mock Activity mockActivity;
+  @Mock PluginRegistry.Registrar mockRegistrar;
+  @Mock BinaryMessenger mockMessenger;
+  @Spy MethodChannel.Result result;
+  @Mock GoogleSignInWrapper mockGoogleSignIn;
+  @Mock GoogleSignInAccount account;
+  private GoogleSignInPlugin plugin;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    when(mockRegistrar.messenger()).thenReturn(mockMessenger);
+    when(mockRegistrar.context()).thenReturn(mockContext);
+    when(mockRegistrar.activity()).thenReturn(mockActivity);
+    plugin = new GoogleSignInPlugin();
+    plugin.initInstance(mockRegistrar.messenger(), mockRegistrar.context(), mockGoogleSignIn);
+    plugin.setUpRegistrar(mockRegistrar);
+  }
+
+  @Test
+  public void requestScopes_ResultErrorIfAccountIsNull() {
```

Comment:
These tests were copied and pasted from the tests in `example/android/app/src/test....`